src/BiDirectionalLinkedList.h

```diff
@@ -0,0 +1,50 @@
+//
+// Created by shvmo on 02/05/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_BIDIRECTIONALLINKEDLIST_H
+#define DATASTRUCTURESWET1_BIDIRECTIONALLINKEDLIST_H
+
+
+#include "BiDirectionalNode.h"
+#include <iostream>
+
+using std::ostream;
+
+template<class Type>
+class BiDirectionalLinkedList {
+public:
+    BiDirectionalLinkedList();
+
+    ~BiDirectionalLinkedList();
+
+    shared_ptr<BiDirectionalNode<Type>> getFirst() const;
+
+    // Done for optimization when we want to iterate a bit faster
+    BiDirectionalNode<Type> *getFirstPointer() const;
+
+    shared_ptr<BiDirectionalNode<Type>> getLast() const;
+
+    void insert(const Type &newValue);
+
+    shared_ptr<BiDirectionalNode<Type>> findNodeByValue(const Type &valueToFind) const;
```

Comment:
we want to allow access to our NODES publicly? 🤨

Reply:
Yes sometimes why not. It can be useful to get the specific pointer so you can remove it more efficiently

Reply:
This function is not used anyway I just did it because it's part of the list interface.

Reply:
Kinda like I purposefully made functions that get a pointer to a node instead of a preallocated value. In order to be able to allocate all memory first then pass allocated smart_ptrs.